File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/trust/AdditionalHostnameVerifier.java
package com.lasthopesoftware.bluewater.client.connection.trust;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLSession;

public class AdditionalHostnameVerifier implements HostnameVerifier {

	private final String additionalHostname;
	private final HostnameVerifier fallbackHostnameVerifier;

	public AdditionalHostnameVerifier(String additionalHostname, HostnameVerifier fallbackHostnameVerifier) {
		this.additionalHostname = additionalHostname;
		this.fallbackHostnameVerifier = fallbackHostnameVerifier;
	}

	@Override
	public boolean verify(String hostname, SSLSession session) {
		return (hostname != null && hostname.equalsIgnoreCase(additionalHostname))
			|| (fallbackHostnameVerifier != null && fallbackHostnameVerifier.verify(hostname, session));
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/trust/AdditionalHostnameVerifier.kt
package com.lasthopesoftware.bluewater.client.connection.trust

import javax.net.ssl.HostnameVerifier
import javax.net.ssl.SSLSession

class AdditionalHostnameVerifier(
    private val additionalHostname: String,
    private val fallbackHostnameVerifier: HostnameVerifier?
) : HostnameVerifier {
    override fun verify(hostname: String?, session: SSLSession?): Boolean {
        return hostname != null && hostname.equals(
            additionalHostname,
            ignoreCase = true
        ) || fallbackHostnameVerifier != null && fallbackHostnameVerifier.verify(
            hostname,
            session
        )
    }
}
